commandlineparser/src/Parser.cpp

```diff
@@ -130,3 +137,85 @@ void cmdParser::Parser::long_help(const std::vector<std::string>&keys) const
 
 	print(keys, print_long_help,"Long_Description");
 }
+
+void cmdParser::Parser::extract_value_as_string(int argc, char**argv)
+{ 
+	std::string key;
+	for (int i = 1; i < argc; i++)
+	{
+		if (!(strcmp(argv[i], "--help") == 0 || strcmp(argv[i], "-h") == 0)&&(!(strcmp(argv[i-1], "--help") == 0 || strcmp(argv[i-1], "-h") == 0)))
+		{
+			if (argv[i][0] == '-' || argv[i][1] == '-')
+			{
+				std::string temp(argv[i]);
+				size_t found = temp.find("=");
+				if (found != std::string::npos)
+				{
+					key = temp.substr(0, found);
+					std::string val = temp.substr(temp.find("=") + 1);
+
+					//validity checking block.
+					if (command_list.find(key) == command_list.end())
+					{
+						throw std::exception(("Entered command [ " + key + " ] not found in the registered command list.").c_str());
+					}
+					else
+					//to remove the old value for already existing key.
+					{
+						command_list.at(key)->clear_the_value();
+					}
+
+					//store for the case --copy=123 so key= --copy and val=123.
+					command_list[key]->set_value(val);
+				}
+
+				//if  only key is there.Example  -cp 142 no '='.
+				else
+				{
+					key = argv[i];
+				
+					//validity checking block for the new key found.
+					if (command_list.find(key) == command_list.end())
+					{
+						throw std::exception(("Entered command [ " + temp + " ] not found in the registered command list.").c_str());
+					}
+					else
+					//remove the old value for already existing key.
+					{
+						command_list.at(key)->clear_the_value();
+					}
+				}
+			}
+
+			//for only data no key. --copy 152 123 25 store such values 152 123 25.
+			else
+			{
+				command_list[key]->set_value(argv[i]);
+			}
+		}
+	}
+}
+
+std::vector <std::string> cmdParser::Parser::getValueAsString(const std::string &  input)const
+{ 
+	//retrive value if the input key is found in command list.
+	if (command_list.find(input) != command_list.end())
+	{
+		// demanding value for command for which user has not given any value from commandline.(later we will set it to default).
+		if (command_list.at(input)->get_value().empty())
```

Comment:
Optimize this.. call get_value() only once

Reply:
done.